### DISCUSSION TITLE
vim (not neovim but vim9) plugin keeps "stealing" my escape key

### DISCUSSION THREAD

**Discussion by vanschelven** [community/community#84035] [Copilot Conversations]
### Select Topic Area

Bug

### Body

I've installed the "neovim plugin" (which is supposedly also a vim plugin) and I've run the setup.

When in insert mode, when I press escape, I expect to go back to normal mode.
However, in some instances (not always) this does not happen, and when I press escape I see some brief flickering after which I'm still in insert mode.

```
vim9 --version
VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Dec 31 2023 11:21:49)
Included patches: 1-2189
```

**Comment by tunahancelik** [comment]
It seems you are encountering an intermittent issue with Neovim where pressing the escape key in insert mode does not always return you to normal mode. Neovim is actually an extension of Vim, designed to improve Vim by adding features, fixing issues, and improving performance. However, it's worth noting that "vim9 --version" suggests you are checking the version of Vim, not Neovim. To check the version of Neovim, you should use the command nvim --version.

i think,

Check Neovim Version: First, ensure you are indeed running Neovim and check its version using nvim --version. Make sure you have the latest version, as this might be a bug that has been fixed in newer releases.

Inspect Key Mappings: It's possible that something in your configuration is remapping the escape key in insert mode. Check your init.vim file (or .vimrc if you are using compatibility mode) for any mappings that could interfere with the escape key. You can also run :verbose imap <Esc> in Neovim to see if the escape key is mapped to something else.

Plugins Conflict: If you have other plugins installed, there might be a conflict. Temporarily disable other plugins and see if the issue persists. If the issue resolves, re-enable the plugins one by one to identify the conflicting plugin.

Check Terminal Configuration: Sometimes, the issue might be with the terminal emulator rather than Neovim itself. Try using a different terminal emulator to see if the problem persists.

Look for Delay: Some configurations or plugins might introduce a delay or require a double press of the escape key. This can happen due to settings like ttimeoutlen in Vim.

Try Neovim in a Clean State: Run Neovim with no configuration file (nvim -u NONE) to see if the issue is related to your personal configuration.

Consult Documentation and Community: Check the Neovim documentation for any known issues or peculiarities with the escape key. Additionally, consider reaching out to the Neovim community (e.g., on GitHub, Reddit, or their official forums) for support.

Report the Bug: If none of these steps work and you suspect this is a bug with Neovim itself, consider reporting it to the Neovim GitHub issue tracker. Be sure to include detailed information about your Neovim version, operating system, terminal emulator, and any relevant configuration details.

**Comment by vanschelven** [comment]
> It seems you are encountering an intermittent issue with Neovim where pressing the escape key in insert mode does not always return you to normal mode.

I am not, as per the title. My problem occurs on _vim_, which is, per the README of the plugin, an installation target as long as I'm on version 9.

**Comment by vanschelven** [comment]
link to plugin docs which state that vim (version 9) is fine:
https://github.com/github/copilot.vim#getting-started

**Comment by vanschelven** [comment]
this issue affected both `vim9` and `nvim`; in the end I drilled it down to mapping `C-[` to a copilot suggestion... but `C-[` is equivalent to escape... so I hijacked myself...
